system/src/system_network_diagnostics.cpp

```diff
@@ -0,0 +1,355 @@
+/**
+ ******************************************************************************
+  Copyright (c) 2013-2015 Particle Industries, Inc.  All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************
+ */
+
+// Mesh signal strength/quality have not been implemented
+#if (PLATFORM_ID != 14) && (PLATFORM_ID != 24)
```

Comment:
Let's prefer to use the macros from `platforms.h`.

We could alternatively change the return values in https://github.com/particle-iot/device-os/blob/develop/wiring/inc/spark_wiring_mesh.h#L59 to negative for now to cause an error to be returned but I agree that it's probably better not to send them at all.

Reply:
Great catch!